packages/core/src/Stores/contract-trade-store.js

```diff
@@ -428,6 +429,12 @@ export default class ContractTradeStore extends BaseStore {
         return length > 0 ? applicable_contracts[length - 1] : {};
     }
 
+    get prev_contract() {
+        const applicable_contracts = this.applicable_contracts();
+        const length = this.contracts[0]?.contract_info.current_spot_time ? applicable_contracts.length : -1;
```

Comment:
can you explain how `current_spot_time` is related to get the prev_contract?

Reply:
same check as for the `last_contract` above. As far as I remember, it's here to make sure this `computed` value gets updated on every update of contract data (every second or two) streaming from API to always have relevant data.